src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -832,20 +867,31 @@ private String constructUserData(boolean graphAlreadyBuilt) {
             else lines.add(String.format("sudo -H -u ubuntu java -jar -Xmx${MEM}k %s/%s.jar --build %s > $BUILDLOGFILE 2>&1", jarDir, jarName, routerDir));
             // Re-upload user data log after build command.
             lines.add(uploadUserDataLogCommand);
-            // Upload the build log file and graph to S3.
+            // Upload the build log file, build report and graph to S3.
             if (!deployment.r5) {
                 String s3BuildLogPath = joinToS3FolderURI(getBuildLogFilename());
+                // upload log file
                 lines.add(String.format("aws s3 cp $BUILDLOGFILE %s ", s3BuildLogPath));
+                // upload report if it was generated
+                String reportPath = String.format("%s/report", routerDir);
+                lines.add(String.format(
+                    "[ -e %s ] && cd %s && zip -r report.zip report && cd - && aws s3 cp %s.zip %s",
+                    reportPath,
+                    routerDir,
+                    reportPath,
+                    joinToS3FolderURI("graph-build-report.zip")
+                ));
+                // upload graph
                 lines.add(String.format("aws s3 cp %s %s ", graphPath, getS3GraphURI()));
             }
         }
         // Determine if graph build/download was successful (and that Graph.obj is not zero bytes).
         lines.add(String.format("FILESIZE=$(wc -c <%s)", graphPath));
         lines.add(String.format("[ -f %s ] && (($FILESIZE > 0)) && GRAPH_STATUS='SUCCESS' || GRAPH_STATUS='FAILURE'", graphPath));
+        // Re-upload user data log before indicating that graph build/download is complete.
+        lines.add(uploadUserDataLogCommand);
         // Create file with bundle status in web dir to notify Data Tools that download is complete.
         lines.add(String.format("sudo echo $GRAPH_STATUS > $WEB_DIR/%s", GRAPH_STATUS_FILE));
-        // Re-upload user data log before final command (and before optional shutdown statement).
-        lines.add(uploadUserDataLogCommand);
```

Comment:
I don't quite understand why this line was moved up.

Reply:
I wanted to make sure that the logs were uploaded before echoing the success into the graph status file. If the DeployJob detects that the graph building is done and terminates the instance before the logs can get uploaded that's not good.